[박소현] week4

ParkSohyunee opened the pull request:
## 요구사항

### 기본

- [x] 이메일 input에서 focus out 할 때, 값이 없을 경우 input에 빨강색 테두리와 아래에 “이메일을 입력해주세요.” 빨강색 에러 메세지가 보이나요?
- [x] 이메일 input에서 focus out 할 때, 이메일 형식에 맞지 않는 값이 있는 경우 input에 빨강색 테두리와 아래에 “올바른 이메일 주소가 아닙니다.” 빨강색 에러 메세지가 보이나요?
- [x] 이메일: test@codeit.com, 비밀번호: codeit101 으로 로그인 시도할 경우, “/folder” 페이지로 이동하나요?
- [x] 비밀번호 input에서 focus out 할 때, 값이 없을 경우 아래에 “비밀번호를 입력해주세요.” 에러 메세지가 보이나요?
- [x] 이외의 로그인 시도의 경우 이메일, 비밀번호 input에 빨강색 테두리와 각각의 input아래에 “이메일을 확인해주세요.”, “비밀번호를 확인해주세요.” 빨강색 에러 메세지가 보이나요?
- [x] 이메일, 비밀번호 input에 에러 관련 디자인을 Components 영역의 에러 케이스로 적용했나요?  

### 심화

- [x] 눈 모양 아이콘 클릭시 비밀번호의 문자열이 보이기도 하고, 가려지기도 하나요?
- [x] 비밀번호의 문자열이 가려질 때는 눈 모양 아이콘에는 사선이 그어져있고, 비밀번호의 문자열이 보일 때는 사선이 없는 눈 모양 아이콘이 보이나요?

## 주요 변경사항

- 로그인 페이지의 이메일, 비밀번호 input 유효성 검증
- 로그인 버튼 클릭시 페이지 이동
- 비밀번호 가리기 기능 구현(공통 파일로 분리)

## 스크린샷
![스크린샷 2023-09-29 오후 7 06 11](https://github.com/codeit-bootcamp-frontend/1-Weekly-Mission/assets/124856726/243c7e7c-da4c-485c-82b9-4e50c2f32828)

![스크린샷 2023-09-29 오후 7 08 47](https://github.com/codeit-bootcamp-frontend/1-Weekly-Mission/assets/124856726/2bb27cbc-013b-4930-ae8b-b719c2be0ccc)

![스크린샷 2023-09-29 오후 7 09 25](https://github.com/codeit-bootcamp-frontend/1-Weekly-Mission/assets/124856726/db050603-5fe2-44a6-8ad9-8f06bae18ea4)

## 멘토에게

- 기능 구현 관련하여 피드백 부탁드립니다. 감사합니다.
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

ParkSohyunee committed:
feat: signin 페이지 이메일, 비밀번호 에러메세지 구현

ParkSohyunee committed:
feat: signin - 이메일 형식 유효성검사 기능 추가

ParkSohyunee committed:
feat: signin - 로그인 기능 추가

ParkSohyunee committed:
feat: signin - 비밀번호 가리기 기능 추가

ParkSohyunee committed:
refactor: signin 리팩토링
- input 에러시 스타일 변경 리팩토링
- 로그인 아이디, 비밀번호 상수 선언
- 'use strict' 추가

ParkSohyunee committed:
refactor: signin - 비밀번호 가리기 기능 로직 개선 및 공통 파일로 분리

llucas93 commented on Linkbrary/src/signin.js:
```diff
@@ -0,0 +1,69 @@
+const emailValidation = (value) => {
+  if (!regExp.test(value)) {
+    emailError.textContent = "올바른 이메일 주소가 아닙니다.";
+    email.classList.add("error");
```
파일전체 element.classList.add() / element.classList.remove() 중복되어 보이는데 별도 함수로 만들면 어떨까요?

```
const addErrorClass = (element) => { ... }
const removeErrorClass = (element) => { ... }
```

ParkSohyunee commented on Linkbrary/src/common/hiddenPassword.js:
```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+const eyeIcons = document.querySelectorAll(".form__input--selected");
```
네 멘토님, 비밀번호 가리기 기능이 향후 회원가입 페이지에도 사용될 때 비밀번호 확인란이 추가되어 요소가 2개가 되므로 querySelectorAll을 사용하였습니다. :) 확인 부탁드립니다.
더불어 다른 피드백 주신 내용 참고하여 리팩토링 진행하겠습니다. 상세히 확인 해 주셔서 감사합니다. 😊

llucas93 replied:
아하 회원가입도 고려하셔서 작성하신거군요 👍👍

ParkSohyunee committed:
refactor: signin page 리팩토링
- ESLint 적용
- input 자동완성 기능 제거
- if 분기문 Early return 패턴 적용
- (기타) 일부 변수명 수정 및 로직 개선

ParkSohyunee committed:
refactor: signin 리팩토링
- focusout 이벤트 로직 개선
- 프리티어 재적용

ParkSohyunee commented:
멘토님 피드백 주신 내용 반영하여 리팩토링 적용 완료하였습니다. 🙂
주요 변경사항은 다음과 같습니다.

- if 분기문 Early return 패턴 적용
- focusout 이벤트 로직 개선 (버블링을 활용하여 이벤트 핸들러를 form 태그에 1번 등록하는 방법으로 개선)
- (기타) 일부 변수명 수정 및 로직 개선
- input 자동완성 기능 제거
- ESLint 적용, 프리티어 재적용
